kroxylicious/src/main/java/io/kroxylicious/proxy/bootstrap/FilterChainFactory.java

```diff
@@ -25,8 +28,23 @@ public class FilterChainFactory {
     private final Configuration config;
 
     public FilterChainFactory(Configuration config) {
-        Objects.requireNonNull(config);
-        this.config = config;
+        this.config = Objects.requireNonNull(config);
+    }
+
+    /**
+     * Validate that a collection of FilterDefinitions contains all the required configuration to configure the filter.
+     * @param filterDefinitions the candidates to validate.
+     * @return the Set of filter names which fail validation.
+     */
+    public static Set<String> validateFilterConfiguration(Collection<FilterDefinition> filterDefinitions) {
+        if (Objects.isNull(filterDefinitions) || filterDefinitions.isEmpty()) {
+            return Set.of();
+        }
+        return filterDefinitions.stream()
+                .filter(f -> f.config() == null)
+                .map(FilterDefinition::type)
+                .filter(type -> ContributionManager.INSTANCE.getDefinition(FilterContributor.class, type).configurationRequired())
```

Comment:
Wonder if we should load the ConfigurationDefinition into the FilterDefinition. I guess we could make this same ContributionManager.INSTANCE.getDefinition from the FilterDefinition constructor. Then this method could change to .filter(f -> f.configurationRequired() && f.config() == null) which would be a little easier on the eye.

Reply:
I don't think we can do it in the `FilterDefinition` ctor as `FilterDefinition` is a record and thus can't have instance variables. 

We can however embed that call inside `FilterDefinition.configurationRequired()`. Which is still easier on the eye. 

I there a preference between two filter clauses or a single compound clause?

Reply:
resolved with 24926c0effe259669bfbdf97422b6e391f2d13fc